xrdp/xrdp_mm.c

```diff
@@ -991,11 +991,185 @@ xrdp_mm_process_rail_drawing_orders(struct xrdp_mm *self, struct stream *s)
+    out_uint32_le(s, 8 + 12);
+    out_uint32_le(s, 16); /* MaxNumMonitors */
+    out_uint32_le(s, 4096); /* MaxMonitorAreaFactorA */
+    out_uint32_le(s, 2048); /* MaxMonitorAreaFactorB */
+    s_mark_end(s);
+    bytes = (int) (s->end - s->data);
+    libxrdp_drdynvc_data(wm->session, chan_id, s->data, bytes);
+    free_stream(s);
+    return 0;
+}
+
+/******************************************************************************/
+static int
+dynamic_monitor_close_response(intptr_t id, int chan_id)
+{
+    LOG(LOG_LEVEL_INFO, "dynamic_monitor_close_response:");
+    return 0;
+}
+
+/******************************************************************************/
+static int
+dynamic_monitor_data_first(intptr_t id, int chan_id, char *data, int bytes,
+                           int total_bytes)
+{
+    LOG(LOG_LEVEL_INFO, "dynamic_monitor_data_first:");
+    return 0;
+}
+
+/******************************************************************************/
+static int
+dynamic_monitor_data(intptr_t id, int chan_id, char *data, int bytes)
+{
+    struct stream ls;
+    struct stream *s;
+    int msg_type;
+    int msg_length;
+    int monitor_index;
+    struct xrdp_process *pro;
+    struct xrdp_wm *wm;
+
+    int MonitorLayoutSize;
+    int NumMonitor;
+
+    int Flags;
+    int Left;
+    int Top;
+    int Width;
+    int Height;
+    int PhysicalWidth;
+    int PhysicalHeight;
+    int Orientation;
+    int DesktopScaleFactor;
+    int DeviceScaleFactor;
+
+    struct xrdp_rect rect;
+    int session_width;
+    int session_height;
+
+    LOG(LOG_LEVEL_INFO, "dynamic_monitor_data:");
+    pro = (struct xrdp_process *) id;
+    wm = pro->wm;
+    g_memset(&ls, 0, sizeof(ls));
+    ls.data = data;
+    ls.p = ls.data;
+    ls.size = bytes;
+    ls.end = ls.data + bytes;
+    s = &ls;
+    in_uint32_le(s, msg_type);
+    in_uint32_le(s, msg_length);
+    LOG(LOG_LEVEL_INFO, "dynamic_monitor_data: msg_type %d msg_length %d",
+           msg_type, msg_length);
+
+    rect.left = 8192;
+    rect.top = 8192;
+    rect.right = -8192;
+    rect.bottom = -8192;
+
+    if (msg_type == 2) /* DISPLAYCONTROL_PDU_TYPE_MONITOR_LAYOUT */
+    {
+        in_uint32_le(s, MonitorLayoutSize);
+        in_uint32_le(s, NumMonitor);
+        LOG(LOG_LEVEL_INFO, "  MonitorLayoutSize %d NumMonitor %d",
+               MonitorLayoutSize, NumMonitor);
+        for (monitor_index = 0; monitor_index < NumMonitor; monitor_index++)
+        {
+            in_uint32_le(s, Flags);
+            in_uint32_le(s, Left);
+            in_uint32_le(s, Top);
+            in_uint32_le(s, Width);
+            in_uint32_le(s, Height);
+            in_uint32_le(s, PhysicalWidth);
+            in_uint32_le(s, PhysicalHeight);
+            in_uint32_le(s, Orientation);
+            in_uint32_le(s, DesktopScaleFactor);
+            in_uint32_le(s, DeviceScaleFactor);
+            LOG(LOG_LEVEL_INFO, "    Flags 0x%8.8x Left %d Top %d "
```

Comment:
LOG_DEVEL(LOG_LEVEL_DEBUG, )

Reply:
Updated.

Reply:
Unfortunately when I make this modification, I get this:

```
xrdp_mm.c:1067:9: error: variable ‘DeviceScaleFactor’ set but not used [-Werror=unused-but-set-variable]
 1067 |     int DeviceScaleFactor;
      |         ^~~~~~~~~~~~~~~~~
xrdp_mm.c:1066:9: error: variable ‘DesktopScaleFactor’ set but not used [-Werror=unused-but-set-variable]
 1066 |     int DesktopScaleFactor;
      |         ^~~~~~~~~~~~~~~~~~
xrdp_mm.c:1065:9: error: variable ‘Orientation’ set but not used [-Werror=unused-but-set-variable]
 1065 |     int Orientation;
      |         ^~~~~~~~~~~
xrdp_mm.c:1064:9: error: variable ‘PhysicalHeight’ set but not used [-Werror=unused-but-set-variable]
 1064 |     int PhysicalHeight;
      |         ^~~~~~~~~~~~~~
xrdp_mm.c:1063:9: error: variable ‘PhysicalWidth’ set but not used [-Werror=unused-but-set-variable]
 1063 |     int PhysicalWidth;
      |         ^~~~~~~~~~~~~
xrdp_mm.c:1058:9: error: variable ‘Flags’ set but not used [-Werror=unused-but-set-variable]
 1058 |     int Flags;
      |         ^~~~~
xrdp_mm.c:1055:9: error: variable ‘MonitorLayoutSize’ set but not used [-Werror=unused-but-set-variable]
 1055 |     int MonitorLayoutSize;
      |         ^~~~~~~~~~~~~~~~~
```

For now, I've moved them to `LOG_LEVEL_DEBUG` and kept the offending log lines as `LOG` -- That may not be the ideal solution we end up with in the end, however.

Reply:
Ok, I've updated this further so that it passes by moving it into a struct.